providers/provider.go

```diff
@@ -85,13 +88,14 @@ type Provider interface {
 	UpdateReplicationGroup(ctx context.Context, instanceID string, params UpdateReplicationGroupParameters) error
 	UpdateParamGroupParameters(ctx context.Context, instanceID string, params UpdateParamGroupParameters) error
 	Deprovision(ctx context.Context, instanceID string, params DeprovisionParameters) error
-	GetState(ctx context.Context, instanceID string) (ServiceState, string, error)
+	ProgressState(ctx context.Context, instanceID string, operation string, primaryNode string, timeOut string) (ServiceState, string, error)
 	GetInstanceParameters(ctx context.Context, instanceID string) (InstanceParameters, error)
 	GetInstanceTags(ctx context.Context, instanceID string) (map[string]string, error)
 	GenerateCredentials(ctx context.Context, instanceID, bindingID string) (*Credentials, error)
 	RevokeCredentials(ctx context.Context, instanceID, bindingID string) error
 	DeleteCacheParameterGroup(ctx context.Context, instanceID string) error
 	FindSnapshots(ctx context.Context, instanceID string) ([]SnapshotInfo, error)
+	TestFailover(ctx context.Context, instanceID string) (string, error)
```

Comment:
This is an asynchronous operation, and I think the name should reflect that. Perhaps “StartFailoverTest”?